lib/si-layer-cache/src/hybrid_cache.rs

```diff
@@ -104,44 +78,24 @@ where
 
         let cache_name: &'static str = config.name.leak();
 
-        let cache: HybridCache<Arc<str>, MaybeDeserialized<V>> = HybridCacheBuilder::new()
-            .with_name(cache_name)
-            .with_metrics_registry(OpenTelemetryMetricsRegistry::new(global::meter(cache_name)))
-            .memory(memory_cache_capacity_bytes)
-            .with_weighter(
-                |_key: &Arc<str>, value: &MaybeDeserialized<V>| match value {
-                    MaybeDeserialized::RawBytes(bytes) => bytes.len(),
-                    MaybeDeserialized::DeserializedValue { size_hint, .. } => *size_hint,
-                },
-            )
-            .storage(Engine::Large)
-            .with_admission_picker(Arc::new(RateLimitPicker::new(
-                config.disk_admission_rate_limit,
-            )))
-            .with_device_options(
-                DirectFsDeviceOptions::new(config.disk_path)
-                    .with_capacity(disk_cache_capacity_bytes),
-            )
-            .with_large_object_disk_cache_options(
-                LargeEngineOptions::new()
-                    .with_buffer_pool_size(config.disk_buffer_size)
-                    .with_eviction_pickers(vec![Box::<FifoPicker>::default()])
-                    .with_flushers(config.disk_buffer_flushers)
-                    .with_recover_concurrency(config.disk_recover_concurrency)
-                    .with_indexer_shards(config.disk_indexer_shards)
-                    .with_reclaimers(config.disk_reclaimers),
-            )
-            .with_recover_mode(RecoverMode::Quiet)
-            .build()
-            .await
-            .map_err(|e| LayerDbError::Foyer(e.into()))?;
+        let cache: MemCache<Arc<str>, MaybeDeserialized<V>> =
+            CacheBuilder::new(memory_cache_capacity_bytes)
+                .with_name(cache_name)
+                .with_metrics_registry(OpenTelemetryMetricsRegistry::new(global::meter(cache_name)))
+                .with_weighter(
+                    |_key: &Arc<str>, value: &MaybeDeserialized<V>| match value {
+                        MaybeDeserialized::RawBytes(bytes) => bytes.len(),
+                        MaybeDeserialized::DeserializedValue { size_hint, .. } => *size_hint,
+                    },
+                )
+                .build();
 
         Ok(Self { cache })
     }
 
     pub async fn get(&self, key: &str) -> Option<V> {
-        match self.cache.obtain(key.into()).await {
-            Ok(Some(entry)) => match entry.value() {
+        match self.cache.get(key) {
```

Comment:
If memory serves this doesn't really change the sematics, but rather a change in API because we're dealing with a `MemCache` rather than a `HybridCache`?

Reply:
Correct, obtain de-dups calls to the disk, so no disk, no obtain.